gcs/innards/proof.hh

```diff
@@ -365,7 +379,8 @@ namespace gcs::innards
          * \name Helpful things for making proof text.
          */
         ///@{
-
+        [[nodiscard]] auto proof_name(const VariableConditionFrom<SimpleOrProofOnlyIntegerVariableID> &) const -> const std::string &;
```

Comment:
Hrm, I'm not sure whether making these public again is a good idea. I guess we could do this for now and then I'll tidy up the API again later?